src/reader/core.py

```diff
@@ -263,18 +264,35 @@ def make_reader(
     # See this comment for details on how it should evolve:
     # https://github.com/lemon24/reader/issues/168#issuecomment-642002049
 
-    storage: StorageType = _storage or Storage(url)
+    # Since StorageType protocol doesn't obligate
+    # to expose read_only property, cannot check
+    # whether passed custom _storage is read_only or not
+    # So if the user provides _storage AND read_only=True --> Exception
+    storage: StorageType
+
+    if _storage is not None:
+        if read_only:  # pragma: no cover
+            raise TypeError("cannot pass read_only together with a custom storage")
+        storage = _storage
+    else:
+        if read_only is None:
+            read_only = False
+        storage = Storage(url, read_only=read_only)
```

Comment:
I don't think is necessary, since `_storage` is undocumented, and will be removed eventually; until then, it's fine for `read_only` to be silently ignored if `_storage` is passed.